### ISSUE TITLE
vscode resolves paths with `..` in them even if the directory doesn't exist / has invalid name

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pylance-release#2226]
### VS Code version

1.63.2

### Extension version

2021.12.1559732655

### OS type

Windows

### OS version

10

### Python distribution

python.org

### Python version

3.10

### Language server

Default

### Expected behaviour

error message because the directories did not exist - it should resolve the path the same way the operating system does:
![image](https://user-images.githubusercontent.com/57028336/147898829-e8722ce3-0eb6-4183-b233-c3ddcb0dbe3f.png)

### Actual behaviour

no error, since it resolves to a directory that does exist: `C:\Users\user\basedtyping\`

this led to extremely confusing behavior when attempting to link to a typeshed path relative to the python `interpreterPath`:

```json
{
    "python.analysis.typeshedPaths": [
        "${command:python.interpreterPath}/../../Lib/site-packages/mypy/typeshed"
     ]
}
```
which seems perfectly valid considering my folder structure is:
```
C:/users/user/
    basedtyping/
        .venv/
            Scripts/
                python.exe
            Lib/
            site-packages/
                mypy/
                    typeshed/
```
but instead the log showed this:
```
typeshedPath c:\Users\user\Lib\site-packages\mypy\typeshed is not a valid directory.
```
because it turns out `python.analysis.typeshedPaths` doesn't resolve `${command:python.interpreterPath}` at all for some reason, so it was just treated as a directory called "${command:python.interpreterPath}", even though that's not even a valid directory name

### Steps to reproduce

1. in `settings.json`, set `python.analysis.typeshedPaths` to a path that doesn't exist:
    ```json
    {
        "python.analysis.typeshedPaths": ["asdf/asdfasdf"]
    }
    ```
2. go to the "output" tab and selete "Python Language Server"
3. see the `typeshedPath` error:
    ```
    typeshedPath c:\Users\user\basedtyping\asdf\asdfasdf is not a valid directory.
    ```
4. set `python.analysis.typeshedPaths` to go back up to a directory that does exist by using `..`
    ```json
    {
        "python.analysis.typeshedPaths": ["asdf/asdfasdf/../.."]
    }
    ```
5. go to the "output" tab and selete "Python Language Server"

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by DetachHead** [comment]
> given the folder structure, I think it will work better if you use the `workspaceFolder` instead of the `command:python.interpreterPath`.

Yeah that’s what I ended up doing for now (it works without the `${workspaceFolder}` part tho because it’s already treated as relative to the workspace)

However that’s not ideal since I intend to commit `settings.json` and there’s no guarantee that everyone else is also using a venv let alone one with that path. Related: https://github.com/microsoft/pylance-release/issues/2221

**Comment by judej** [comment]
Closing old issue. If this is still a problem, please reopen with the information requested. thanks

**Comment by DetachHead** [comment]
@judej what information was requested that I didn't provide? Also I don't have permission to re-open it

**Comment by debonte** [comment]
@DetachHead, sorry about that, seems we closed this incorrectly when cleaning up old issues.

**Comment by rchiodo** [comment]
Thanks for the report. We investigated the current behavior.

`${command.python.interpreterPath}` is not valid VS Code command-variable syntax (the command form uses a colon), and the syntactically valid `${command:python.interpreterPath}` is not expanded in Pylance path settings such as `python.analysis.typeshedPaths`. Unresolved command variables are rejected by current Pylance, so the exact original example no longer resolves to the path shown in the report.

For static paths, `..` components are normalized lexically before the resulting path is checked on disk. This means a canceled intermediate component is not independently validated. That is expected path-normalization behavior, and we don't plan to change it.

For portable configuration, please use a workspace-relative path, `${workspaceFolder}`, or a supported environment variable. Supporting an interpreter-relative substitution would be a separate feature request rather than a change to `..` handling.

Closing this issue as not planned.
